### PR TITLE
rose app-run: Implement ROSE_APP_COMMAND_KEY

### LINKED ISSUE #1934: rose app-run: command key environment variable override
A user writes:

> When invoking `rose app-run`, I often use the `ROSE_APP_OPT_CONF_KEYS` environment variable to control the optional configuration. I use this rather than the `–O` switch to the command itself as it is more convenient/configurable that having to change the `command scripting` entry for the task in cylc.
> 
> Would it be possible to have an environment variable that does the same for the command key (`-c`, `--command-key`) option too? Something like `ROSE_APP_OPT_COMMAND_KEY`, perhaps?

I've initially put this in 'later' but it is only a small amount of work, so we could do it sooner...

### PR DESCRIPTION
**Pull request by arjclark**
Closes #1934 

Adds `ROSE_APP_COMMAND_KEY` for equivalence to `--command-key` option.

Includes:
 * doc changes
 * added test

@matthewrmshin - please review 1
@oliver-sanders - please review 2

### CODE REVIEW

**Comment by arjclark** [comment]
Travis test battery failures here due to cylc/cylc#2067 changes (works fine against cylc 6.11.2 so manual testing against that advised until #2021 goes in).

**Comment by arjclark** [comment]
Now with #2021 changes in. Battery should now pass.

File: doc/rose-rug-task-run.html [hunk 1]
```diff
@@ -317,6 +317,11 @@ <h3 id="rose-task-run.command">Command Selection</h3>
               task-run</code>, the arguments are run as a shell
               command.</li>
 
+              <li>If the <code>ROSE_APP_COMMAND_KEY</code> environment variable
+              is set, the command specified in the <var>[command]KEY</var>
+              setting in the application configuration whose KEY matches it
+              is used.</li>
```
Comment (oliver-sanders) [inline doc/rose-rug-task-run.html]: Should the environment variable come third to allow the CLI to selectively overwrite a general setting?

Comment (arjclark) [inline doc/rose-rug-task-run.html]: Ordering adjusted in latest commit (makes sense to me that way round) - if @matthewrmshin is happy should be good to go.

**Pull request merged**